ReSharper.FSharp/src/FSharp/FSharp.Psi.Daemon/src/Analyzers/RecordExprAnalyzer.fs

```diff
@@ -64,23 +65,34 @@ type RecordExprAnalyzer() =
                         let currentFieldNameReversed = currentFieldBinding.ReferenceName |> appendFieldNameReversed previousFieldNameReversed
                         ValueSome(rootFieldBinding, currentFieldNameReversed)
                 else
-                    produceHighlighting fieldsChainMatch previousFieldBinding consumer
+                    if isSingleLine then produceHighlighting fieldsChainMatch previousFieldBinding consumer
                     ValueNone
 
             match currentFieldBinding.Expression.IgnoreInnerParens() with
             | :? IRecordExpr as recordExpr ->
-                collectRecordExprsToSimplify recordExpr currentFieldBinding copyExpr fieldsChainMatch consumer
+                let nextFieldIdx = i + 1
+                let nextFieldOnTheSameLine =
+                    if nextFieldIdx = fieldBindings.Count then false else
+                    fieldBindings[nextFieldIdx].StartLine = currentFieldBinding.EndLine
+
+                if fieldsChainMatch.IsNone && not isSingleLine && nextFieldOnTheSameLine then
+                    startNewAnalyze recordExpr consumer
+                else
+                    collectRecordExprsToSimplify recordExpr currentFieldBinding copyExpr fieldsChainMatch consumer
             | _ ->
-                if not searchInDepthWhileMatched then () else
+                if not searchInDepthWhileMatched || not isSingleLine then () else
                 produceHighlighting fieldsChainMatch currentFieldBinding consumer
 
+    and startNewAnalyze recordExpr consumer =
```

Comment:
It's difficult to understand what does the function do, from the `startNewAnalyze` name. Is there some other function that ends it? `Analyze` is not a noun, what does the name actually mean?

Could it be something simple, while descriptive, like `analyzeRecordExpr`?